core/frontend/src/RealityDataConnection.ts

```diff
@@ -0,0 +1,118 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { GuidString } from "@bentley/bentleyjs-core";
+import { RealityDataSourceKey, RealityDataSourceProps } from "@bentley/imodeljs-common";
+import { RealityData, RealityDataClient } from "@bentley/reality-data-client";
+import { AuthorizedFrontendRequestContext } from "./FrontendRequestContext";
+import { RealityDataSource, realityDataSourceKeyToString } from "./RealityDataSource";
+
+/**
+ * This interface provide methods used to access a reality data from ContextShare
+ * @alpha
+ */
+export interface IRealityDataConnection {
+  getRealityData(): RealityData | undefined;
+  getRealityDataType(): string | undefined;
+  getServiceUrl(iTwinId: GuidString | undefined): Promise<string | undefined>;
+  getSource(): RealityDataSource;
+}
+
+/**
+ * This class encapsulates access to a reality data from ContextShare
+ * There is a one to one relationship between a reality data and the instances of present class.
```

Comment:
If these RealityDataConnections are going to be cached and shared then they (and their properties, like RealityData) should be immutable.